interfaces/Portal/src/app/services/api.service.ts

```diff
@@ -38,215 +38,128 @@ export class ApiService {
-        })
-        .catch((err) => reject(err)),
-    );
+    return this.performRequest('POST', endpoint + path, {
+      anonymous,
+      responseAsBlob,
+      params,
+      body,
+      isUpload,
+    });
   }
 
-  put(endpoint: string, path: string, body: object): Promise<any> {
-    const security = this.showSecurity(false);
-    console.log(`ApiService PUT: ${security} ${endpoint}${path}`, body);
-
-    return new Promise((resolve, reject) =>
-      this.http
-        .put(endpoint + path, body, {
-          headers: this.createHeaders(false),
-          withCredentials: true,
-        })
-        .pipe(
-          tap((response) =>
-            console.log(
-              `ApiService PUT: ${security} ${endpoint}${path}:`,
-              body,
-              '\nResponse:',
-              response,
-            ),
-          ),
-          catchError((error: HttpErrorResponse): Observable<any> => {
-            return this.handleError(error);
-          }),
-        )
-        .toPromise()
-        .then((response) => {
-          if (response && response.error) {
-            throw response;
-          }
-          return resolve(response);
-        })
-        .catch((err) => reject(err)),
-    );
+  put(endpoint: string, path: string, body: unknown): Promise<any> {
+    return this.performRequest('PUT', endpoint + path, {
+      body,
+    });
   }
 
   patch(
     endpoint: string,
     path: string,
-    body: object,
+    body: unknown,
     responseAsBlob = false,
     isUpload = false,
     params: HttpParams = null,
   ): Promise<any> {
-    const security = this.showSecurity(false);
-    console.log(`ApiService PATCH: ${security} ${endpoint}${path}`, body);
-
-    return new Promise((resolve, reject) =>
-      this.http
-        .patch(endpoint + path, body, {
-          headers: this.createHeaders(isUpload),
-          responseType: responseAsBlob ? 'blob' : null,
-          withCredentials: true,
-          params,
-        })
-        .pipe(
-          tap((response) =>
-            console.log(
-              `ApiService PATCH: ${security} ${endpoint}${path}${
-                params ? `\nParams ${params}` : ''
-              }:`,
-              body,
-              '\nResponse:',
-              response,
-            ),
-          ),
-          catchError((error: HttpErrorResponse): Observable<any> => {
-            return this.handleError(error);
-          }),
-        )
-        .toPromise()
-        .then((response) => {
-          if (response && response.error) {
-            throw response;
-          }
-          return resolve(response);
-        })
-        .catch((err) => reject(err)),
-    );
+    return this.performRequest('PATCH', endpoint + path, {
+      body,
+      responseAsBlob,
+      isUpload,
+      params,
+    });
   }
 
   delete(
     endpoint: string,
     path: string,
-    body?: object,
-    params: HttpParams = null,
+    body?: unknown,
+    params?: HttpParams,
```

Comment:
The other method-methods have the format: `params: HttpParams = null,`, would this be better for all?